cmd/gapir/cc/main.cpp

```diff
@@ -59,25 +59,29 @@ using namespace gapir;
 
 namespace {
 
+std::shared_ptr<MemoryAllocator> createAllocator() {
+  size_t size =
+      (sizeof(void*) < 8 ? 2ull : 16ull) * 1024ull * 1024ull * 1024ull;
+  std::shared_ptr<MemoryAllocator> allocator;
+
+  while (allocator == nullptr) {
+    try {
+      allocator = std::shared_ptr<MemoryAllocator>(new MemoryAllocator(size));
```

Comment:
Should this be `allocator = MemoryAllocator::create(size)` ?

And do you actually get a null pointer here if the malloc fails?

Reply:
You get an exception and exit the try block without assigning anything to the lhs. See comment above!

Reply:
huh, I thought we had exceptions turned off in gapir.
I just noticed this, and would strongly prefer we avoid exceptions. They have a runtime cost, and since gapir is trying to be a very fast VM, we should probably try to avoid them.

Reply:
https://google.github.io/styleguide/cppguide.html#Exceptions << Style guide forbids them as well.